lib/ws_servers/api/algos/algo_worker.js

```diff
@@ -310,6 +319,14 @@ class AlgoWorker {
       throw new Error('Failed to start algo order')
     }
   }
+
+  checkPermissions (caps) {
+    const hasAllRequiredPermissions = requiredPermissions.every(perm => _get(caps, perm))
+
+    if (!hasAllRequiredPermissions) {
+      throw new Error('The given API key does not have the required permissions, please make sure to enable "Get orders and statuses", "Create and cancel orders", "Get wallet balances and addresses"')
```

Comment:
it would be good  to replace the exception with a boolean and throw the exception at the highest level

Reply:
done